app/src/main/java/com/lorenzovainigli/foodexpirationdates/view/composable/screen/SettingsScreen.kt

```diff
@@ -111,10 +121,45 @@ fun SettingsScreen(
             .padding(10.dp),
         verticalArrangement = Arrangement.spacedBy(16.dp)
     ) {
+
         Text(
             text = stringResource(R.string.behaviour),
             style = MaterialTheme.typography.labelLarge
         )
+
+        SettingsItem(
+            label = stringResource(R.string.enable_screen_protection),
+            description = stringResource(R.string.protect_your_screen)
+        ) {
+            Row(
+                verticalAlignment = Alignment.CenterVertically,
+                modifier = Modifier.padding(start = 11.dp)
+            ) {
+                Switch(
+                    modifier = Modifier.padding(start = 4.dp),
+                    checked = isScreenProtectionEnabled,
+                    onCheckedChange = { enabled ->
+                        isScreenProtectionEnabled = enabled
+
+                        setScreenProtectionEnabled(context, enabled)
+
+                        if (enabled) {
+                            (context as Activity).window.setFlags(
+                                WindowManager.LayoutParams.FLAG_SECURE,
+                                WindowManager.LayoutParams.FLAG_SECURE
+                            )
+                            Log.d("isScreenProtectionEnabled", "Screen protection enabled: true")
+                        } else {
+                            (context as Activity).window.clearFlags(WindowManager.LayoutParams.FLAG_SECURE)
+                            Log.d("isScreenProtectionEnabled", "Screen protection enabled: false")
+                        }
+                    }
+                )
+                Spacer(modifier = Modifier.width(8.dp))
+                Text(text = if (isScreenProtectionEnabled) "Enabled" else "Disabled")
```

Comment:
Do you think this is necessary? I think that the switch don't need a label, like the one for dynamic colors.
If you agree, please remove this `Text` item.